lib/promscrape/scrapework.go

```diff
@@ -305,6 +305,8 @@ func (sw *scrapeWork) scrapeInternal(scrapeTimestamp, realTimestamp int64) error
 		wc.resetNoRows()
 		up = 0
 		scrapesSkippedBySampleLimit.Inc()
+		err = fmt.Errorf("the response from %q exceeds sample_limit=%d; "+
```

Comment:
I think it would be better just logging this error here with `logger.Warnf()` instead of forwarding it to the caller, since this is expected error and it is fully handled here.

Reply:
When a scrape fails, I would like to see the exact reason for the error on the /targets & /api/v1/targets pages.
In my company not everyone has access to the vmagent stderr, but it is possible for them to see the /targets webpage.